### DISCUSSION TITLE
A ranked beatmap has background video playing too early (Lagtrain)

### DISCUSSION THREAD

**Discussion by TechniKris** [ppy/osu#29191] [Q&A]
This beatmap https://osu.ppy.sh/b/2722655 has its video playing too early. It's very noticeable, because, through the way it's mapped, it makes the player cursordance together with the video.

Issue doesn't occur on stable. Here's a video comparing stable and lazer. I synced up gameplay as close as possible, but on lazer the video is playing 8 frames (of this 60fps recording) too early.

https://github.com/ppy/osu/assets/30842158/25a110eb-f6e2-4c49-aab8-f2814283ee52

The video was recorded a while ago, but I confirmed that the issue still occurs on newest releases of lazer and does not occur on stable.

I have already reported this issue in a form of a comment here https://github.com/ppy/osu/issues/13696#issuecomment-1584094285 but perhaps that issue was unrelated to mine, since that one got fixed, yet the video in this beatmap is still off.
Now I decided to make it a discussion instead of an issue, because I don't know what's the culprit, ~~or if it's even an issue that affects anyone else besides my setup~~ _I have confirmed it also happens to another person on Windows._ I don't know whether it affects any other map either.

Here are logs from my system. I started up the game, played a replay on he problematic map, then exported logs.
[compressed-logs.zip](https://github.com/user-attachments/files/16417210/compressed-logs.zip)

**Comment by peppy** [comment]
I've checked that the 29 ms specified video offset is being applied correctly. There may be an issue on stable's end which means that lazer is better aligned than stable, not sure. The misalignment isn't enough to matter in most cases, though.

**Reply by TechniKris** [reply]
If it's stable applying the offset incorrectly, perhaps said offset should also be applied in lazer, at least for cases where the map is made with stable's offset in mind. It's definitely a low priority thing, but it's still a small discrepancy that could be considered buggy behavior.

I apologize for the late reply, I must've missed the notification about your answer 😅

**Comment by Tenexxt** [comment]
Well, it is called lag train.
Jokes aside, I will check if this also happens on my end, and on what settings

**Reply by Tenexxt** [reply]
In comparison with stable, and the overall flow of the map i can tell the video is appearing too fast on Direct 3D 11

With OpenGL, Vulkan and Exp. Direct 3D 11 its the same offset as before, aka the same as in the video.

I also tested on Android and got the same results (so i now know its not my pc working the game too fast, because my phone is getting fried from osu lol)

The results are the same no matter the renderer, device (unless its just Windows and Android having this, but what are the odds for that), whatever the game is running on multithreading or on a single thread, fps and if hardware acceleration is on/off for videos.

So its a 100% game issue and not device, and since the video is too early, it maybe has something with how lazer handles the starting points of videos, or maybe its just the optimisation of the game being so better than stable that the offsets are now being shifted.

A way to fix this would be, well, finding what is lazer doing to render the videos too fast, but if it can't be found, the the user could just be given an option to offset the Videos and/or storyboard for themselfs, kinda like with audio offset, exept not needed 99% of the time and would only have a use if a mapper decides to map to the video (so most likely, inabakumori maps only -_-)

**Reply by TechniKris** [reply]
I assumed it wouldn't be related to the renderer or the platform, hence I didn't initially list it, but for the sake of completeness: I've tried it on Linux with all available renderers: default OpenGL, Veldrid OpenGL and Vulkan, and the issue was there in all cases.
